package-versions/packageversion.md

```diff
@@ -1,105 +1,95 @@
-    <PackageReference Include="Moq" Version="4.10.0" />
-    <!-- Version 2.8.0 defined in the Version attribute -->
-    <PackageReference Include="Serilog" Version="2.8.0" />
-  </ItemGroup>
-</Project>
-```
-
-## Enable Central Package Version Management
-Once a `Directory.Packages.props` file has been created, add a `<ManagePackageVersionsCentrally>` item set to `true` in the `<PropertyGroup>` in a `Directory.Build.props` file.
-
-#### Directory.Build.props
-```xml
-<Project>
-  <PropertyGroup>
-    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
-  </PropertyGroup>
-</Project>
-```
-
-To date, our engineering efforts for CPVM have been focused on supporting restore operations for development and CI/CD pipelines. In the future, we are planning to add additional tooling for Visual Studio.
-
-## Runtime Targeting with .NET 5 SDK
-Updates to CPVM will only be included in the .NET 5 SDK. You can target .NET Core and .NET Framework from the .NET 5 SDK.
-
-.NET Core runtimes and SDK's are decoupled, but they share the same major and minor versions. Any version of the .NET Core or .NET 5 SDK can build any .NET Framework app or any .NET Core app that is equal to or lower than the version of the SDK installed if the relevant Targeting Pack is installed.
-
-Currently, using the .NET 5 SDK with Visual Studio requires installing the preview version of Visual Studio. In the future when .NET 5 is released, it will be possible to install the .NET 5 SDK in the General Availability channel of Visual Studio version 16.8.
-
-## What works today?
-- Visual Studio 16.7 command line restore operations
-- Visual Studio GUI restore operations
-
-## What doesn't work today?
-- Adding a package via the NuGet package management interface in Visual Studio does not currently support CPVM. Once a package has been added using a GUI, it is necessary to add a corresponding <PackageVersion> entry within Directory.Packages.props.
-- Similarly, modifying a package version using the Visual Studio package management interface will update the project file, and it is necessary to move this Version="x.x.x" attribute to the corresponding <PackageVersion> entry within Directory.Packages.props.
+---
+layout: default
+title: Package Versions
+nav_order: 2
+has_children: true
+---
+
+# Central Package Version Management
+Central Package Version Management allows every project within a solution to source their NuGet package versions from a single `Directory.Packages.props` file at the solution level.
+
+## Managing Package Versions at the Solution Level
+
+Managing dependencies across many projects, solutions, and repositories can be very difficult at scale as `<PackageReference>` can be defined in project files, *.props files, and *.targets files throughout the code.
+
+By centrally managing package versions, solutions can maintain more consistency in their NuGet dependencies across projects.
+
+## Transitive Dependencies
```

Comment:
This section does not address a common problem we have with transitive dependencies, which is:

Let A depend on B.
Suppose the PackageVersion for A exists. B does not have a defined version.
I add a PackageReference to B.
I do _not_ want to have to specify a version for B. It should implicitly come from the version specified by A.

Real world scenario: MessagePack depends on MessagePack.Annotations. Almost no one cares about this Annotations dependency nor references it explicitly. But if they do, they should get the one brought in by the main parent package with no further effort.
In particular I do _not_ want to see MessagePack and MessagePack.Annotations *both* have to specify the version explicitly or else we have an update problem where someone updates the version for one and not the other and then we have version shear where either we get a bad combination of binaries or nuget restore fails because a dependency is accidentally lowered for a project when MessagePack.Annotations isn't updated but MessagePack is.

Reply:
I'm not sure if I'm understanding this correctly, but this sounds like the case where no `<PackageVersion>` is specified for the  transitive dependency.

This behavior already works as you described today, but I did not define it here in the document.

Would it help if I clarify that if no `<PackageVersion>` is defined and a single `<PackageReference>` is importing a transitive dependency that the version specified by the package for the transitive dependency is imported?

Reply:
> Would it help if I clarify that if no `<PackageVersion>` is defined and a single `<PackageReference>` is importing a transitive dependency that the version specified by the package for the transitive dependency is imported?

I don't fully understand the second half of your question, and I'm not sure we're thinking of the same thing. What I'd like to see added to the docs (and be true in the product) is this:

> When a `PackageReference` refers to a package for which no `PackageVersion` is defined, but that same package also appears as a transitive dependency of another package that _does_ have a `PackageVersion` specified, then the version used for that `PackageReference` will implicitly come from the version required via the existing transitive dependency.

Or in other words (if this is easier to understand):

> When `PackageVersion` for package ID `A` exists and `A` depends on `B`, a `PackageReference` may be added to `A` or `B`. If no `PackageVersion` for `B` is specified, the version for `B` will implicitly come from from the version required by `A` as expressed in `A`'s dependencies.